litestar/testing/client/async_client.py

```diff
@@ -156,7 +156,7 @@ async def request(
             auth=auth,
             follow_redirects=follow_redirects,
             timeout=timeout,
-            extensions=extensions,
+            extensions=None if extensions is None else dict(extensions),
```

Comment:
The reason for this, and all the others like it, is that we have typed the `extensions` param to the method `Mapping`, however httpx has typed it `MutableMapping` in their method that we call through.

Reply:
Do we prefer runtime overhead or `type: ignore` in this case?

Reply:
The thing that swayed me is that its not technically safe if we allow an immutable mapping into our method and pass it through where a mutable one is expected, and being a test client I figure that the performance implications aren't significant.

However, I also get that there is likely to be few, if not zero occurrences of users actually passing an immutable mapping through to this in the wild. But if that is the argument that sways, we should probably just change our methods to `MutableMapping` also.

`httpx` only changed this from `Mapping` to `MutableMapping` a month ago - they did this on a minor version bump, but they are also zero-ver ATM.